Tests/scripts/common.py

```diff
@@ -264,63 +265,65 @@ def person_in_charge(commit):
 
-    Returns:
-        boolean , the commit that triggered the current pipeline
-    """
-    current_pipeline_commit_timestamp = None
-    previous_pipeline_commit_timestamp = None
-    the_triggering_commit = None
-    for commit in commits:
-        if commit.id == previous_pipeline_sha:
-            previous_pipeline_commit_timestamp = parser.parse(commit.created_at)
-        if commit.id == current_pipeline_sha:
-            current_pipeline_commit_timestamp = parser.parse(commit.created_at)
-            the_triggering_commit = commit
-    if not current_pipeline_commit_timestamp or not previous_pipeline_commit_timestamp:
-        return False, None
-    return current_pipeline_commit_timestamp > previous_pipeline_commit_timestamp, the_triggering_commit
-
-
-def is_pivot(single_pipeline_id, list_of_pipelines, commits):
-    """
-    Check if a given pipeline is a pivot,
-    i.e. if the previous pipeline was successful and the current pipeline failed and vise versa
-   Args:
-    single_pipeline_id: gitlab pipeline ID
-    list_of_pipelines: list of gitlab pipelines
-    commits: list of gitlab commits
-    Return:
-        boolean | None, gitlab commit object| None
-    """
-    current_pipeline = next((pipeline for pipeline in list_of_pipelines if pipeline.id == int(single_pipeline_id)), None)
-    pipeline_index = list_of_pipelines.index(current_pipeline) if current_pipeline else 0
-    if pipeline_index == 0:         # either current_pipeline is not in the list , or it is the first pipeline
-        return None, None
-    previous_pipeline = list_of_pipelines[pipeline_index - 1]
-
-    # if previous pipeline was successful and current pipeline failed, and current pipeline was created after
-    # previous pipeline (n), then it is a negative pivot
-    if current_pipeline:
-        in_order, pivot_commit = are_pipelines_in_order_as_commits(commits, current_pipeline.sha, previous_pipeline.sha)
-        if in_order:
-            if previous_pipeline.status == 'success' and current_pipeline.status == 'failed':
-                return True, pivot_commit
-            if previous_pipeline.status == 'failed' and current_pipeline.status == 'success':
-                return False, pivot_commit
-    return None, None
+
+def is_pivot(current_pipeline, previous_pipeline):
+
+    in_order = are_pipelines_in_order(current_pipeline, previous_pipeline)
+    if in_order:
+        if previous_pipeline.status == 'success' and current_pipeline.status == 'failed':
+            return True
+        if previous_pipeline.status == 'failed' and current_pipeline.status == 'success':
+            return False
+    return None
+
+
+def get_reviewer(pr_url: str) -> str | None:
+    approved_reviewer = None
+    try:
+        # Extract the owner, repo, and pull request number from the URL
+        parts = pr_url.split("/")
+        repo_owner = parts[-4]
+        repo_name = parts[-3]
+        pr_number = parts[-1]
```

Comment:
```suggestion
        repo_owner, repo_name, _, pr_number = pr_url.split("/")
```

Reply:
not as neet, but ok.
_1, _2,_3, repo_owner, repo_name,_4, pr_number = pr.split("/")